provider/core/src/refcount.rs

```diff
@@ -0,0 +1,222 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use alloc::boxed::Box;
+use core::fmt::Debug;
+use core::mem::ManuallyDrop;
+use core::ptr::NonNull;
+use yoke::CloneableCart;
+
+#[cfg(not(feature = "sync"))]
+pub(crate) use alloc::rc::Rc as SelectedRc;
+#[cfg(feature = "sync")]
+pub(crate) use alloc::sync::Arc as SelectedRc;
+
+type WrappedType = Box<[u8]>;
+
+const SENTINEL_PTR: *mut WrappedType = core::mem::align_of::<WrappedType>() as *mut WrappedType;
```

Comment:
Did this in https://github.com/unicode-org/icu4x/pull/4449

Reply:
so are you closing this in favour of that?

Reply:
Haven't cleared it with @Manishearth yet who said yesterday that it might be more appropriate for provider core but I put up #4449 to make a case that it fits with yoke. In the process of porting the code I made some cleanups.